src/realm/sync/protocol.hpp

```diff
@@ -197,28 +197,38 @@ struct CompensatingWriteErrorInfo {
     std::string reason;
 };
 
+struct ResumptionDelayInfo {
+    std::chrono::milliseconds max_resumption_delay_interval = std::chrono::minutes{5};
+    std::chrono::milliseconds resumption_delay_interval = std::chrono::seconds{1};
+    int resumption_delay_backoff_multiplier = 2;
+};
+
 struct ProtocolErrorInfo {
     ProtocolErrorInfo() = default;
-    ProtocolErrorInfo(const std::string& msg, bool do_try_again)
-        : message(msg)
+    ProtocolErrorInfo(int error_code, const std::string& msg, bool do_try_again)
+        : error_code(error_code)
+        , message(msg)
         , try_again(do_try_again)
         , client_reset_recovery_is_disabled(false)
         , should_client_reset(util::none)
     {
     }
+    int error_code = 0;
```

Comment:
This is shadowed by SessionErrorInfo::error_code which is going to be a bit confusing. How about calling it raw_error_code or something else?

Reply:
Fixed.